src/CashIn/Internal/AddNullableAmountConverter.cs

```diff
@@ -0,0 +1,16 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace X.Paymob.CashIn.Internal;
+
+internal sealed class AddNullableAmountConverter : JsonConverter<string> {
+    public override string Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) {
+        return reader.TokenType.Equals(JsonTokenType.Number) && reader.TryGetDecimal(out decimal result)
```

Comment:
you don't need to check `reader.TokenType`, since `reader.TryGetDecimal` will do that for you.